packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -426,21 +428,38 @@ contract L2Genesis is Deployer {
             return;
         }
 
-        GovernanceToken token = new GovernanceToken();
+        address token;
+        if (cfg.useInterop()) {
+            token = address(new GovernanceTokenInterop());
+        } else {
+            token = address(new GovernanceToken());
+        }
         console.log("Setting %s implementation at: %s", "GovernanceToken", Predeploys.GOVERNANCE_TOKEN);
-        vm.etch(Predeploys.GOVERNANCE_TOKEN, address(token).code);
+        vm.etch(Predeploys.GOVERNANCE_TOKEN, token.code);
 
         bytes32 _nameSlot = hex"0000000000000000000000000000000000000000000000000000000000000003";
         bytes32 _symbolSlot = hex"0000000000000000000000000000000000000000000000000000000000000004";
         bytes32 _ownerSlot = hex"000000000000000000000000000000000000000000000000000000000000000a";
 
-        vm.store(Predeploys.GOVERNANCE_TOKEN, _nameSlot, vm.load(address(token), _nameSlot));
-        vm.store(Predeploys.GOVERNANCE_TOKEN, _symbolSlot, vm.load(address(token), _symbolSlot));
+        vm.store(Predeploys.GOVERNANCE_TOKEN, _nameSlot, vm.load(token, _nameSlot));
+        vm.store(Predeploys.GOVERNANCE_TOKEN, _symbolSlot, vm.load(token, _symbolSlot));
         vm.store(Predeploys.GOVERNANCE_TOKEN, _ownerSlot, bytes32(uint256(uint160(cfg.governanceTokenOwner()))));
 
         /// Reset so its not included state dump
-        vm.etch(address(token), "");
-        vm.resetNonce(address(token));
+        vm.etch(token, "");
+        vm.resetNonce(token);
+    }
+
+    /// @notice This predeploy is following the safety invariant #2.
+    function setGovernanceDelegation() public {
+        if (!cfg.enableGovernance()) {
+            console.log("Governance not enabled, skipping setting GovernanceDelegation");
+            return;
+        }
+
+        console.log("Setting %s implementation at: %s", "GovernanceDelegation", Predeploys.GOVERNANCE_DELEGATION);
```

Comment:
Did you mean to not put this behind a proxy? It will need to be proxied based on its address, existing chains already have a proxy at the defined address

Reply:
ah gotcha, so govtoken should be proxied? currently it's not proxied so wasn't sure of changing the behaviour

Reply:
`GovernanceDelegation` is different than `GovToken`